homeassistant/components/sensor/zha.py

```diff
@@ -82,8 +85,11 @@
 
     if discovery_info['new_join']:
         cluster = list(in_clusters.values())[0]
+        report_attribute = sensor.value_attribute
+        if hasattr(sensor, 'report_attribute'):
+            report_attribute = sensor.report_attribute
         await helpers.configure_reporting(
-            sensor.entity_id, cluster, sensor.value_attribute,
+            sensor.entity_id, cluster, report_attribute,
```

Comment:
IMO we should be reporting both "battery_voltage" attr id 0x0020 and "battery_percentage_remaining" attr id 0x0021 as some vendors actually do support battery % remaining (like SmartThings/Samji sensors) and we could get that directly without zigpy quirks

Reply:
or maybe it is too late for me to comment, but on the 3rd iteration I think we do need to report both attributes.

Reply:
which is a breath to implement with https://github.com/home-assistant/home-assistant/pull/19177 merged :)

Reply:
Agree on attr reporting. That said, handlers are almost guaranteed necessary because the battery formulas are all over the place